src/Dapr.AspNetCore/CloudEventsMiddleware.cs

```diff
@@ -85,8 +85,19 @@ private async Task ProcessBodyAsync(HttpContext httpContext, string charSet)
             // Check whether to use data or data_base64 as per https://github.com/cloudevents/spec/blob/v1.0.1/json-format.md#31-handling-of-data
             var isDataSet = json.TryGetProperty("data", out var data);
             var isBinaryDataSet = json.TryGetProperty("data_base64", out var binaryData);
+            var isEntrySet = json.TryGetProperty("entries", out var entry);
 
-            if (isDataSet && isBinaryDataSet)
+            // An "entries" field in the payload represents a bulk publish request so we handle the entire payload
+            // as it is and don't do anything
+            if (isEntrySet)
+            {
+                body = new MemoryStream();
+                await JsonSerializer.SerializeAsync<JsonElement>(body, json);
```

Comment:
How does this look as an actual payload coming in? My hope is that it would be an array of the actual events so we don't need the `CloudEventModel` in the middle as that's the point of this middleware is to remove the cloudevent.

Below we just reference the `data` field for example.

Reply:
Discussed offline and addressed. The reason being, if we just reference `data`, we might not get other information like `entryId`. These fields might be important as they tell the consumer which message failed/passed.